jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -30,39 +32,51 @@ class JobRunExecutorActor(
   taskTracker:                TaskTracker,
   driverHolder:               MarathonSchedulerDriverHolder,
   clock:                      Clock,
-  val behavior:               Behavior) extends Actor with Stash with ActorLogging with ActorBehavior {
+  val behavior:               Behavior)(implicit scheduler: Scheduler) extends Actor with Stash with ActorLogging with ActorBehavior {
+
   import JobRunExecutorActor._
   import JobRunPersistenceActor._
   import TaskStates._
+  import context.dispatcher
+
+  private[impl] var startingDeadlineTimer: Option[Cancellable] = None
 
   lazy val persistenceActor = persistenceActorRefFactory(run.id, context)
   var jobRun: JobRun = run
 
   val runSpecId = jobRun.id.toPathId
 
   override def preStart(): Unit = {
-    jobRun.status match {
-      case JobRunStatus.Initial => becomeCreating()
-
-      case JobRunStatus.Starting | JobRunStatus.Active =>
-        launchQueue.get(runSpecId) match {
-          case Some(info) if info.finalTaskCount > 0 =>
-            log.info("found an active launch queue for {}, not scheduling more tasks", runSpecId)
-            val tasks = taskTracker.appTasksLaunchedSync(runSpecId).collect {
-              // FIXME: we do currently only allow non-resident tasks. Since there is no clear state on
-              // Marathon's task representation, this is the safest conversion we can do for now:
-              case task: LaunchedEphemeral => JobRunTask(task)
-              case task: Task              => throw new UnexpectedTaskState(task)
-            }
-            self ! Initialized(tasks)
-
-          case _ =>
-            self ! Initialized(tasks = Nil)
-        }
-
-      case JobRunStatus.Success => becomeFinishing(jobRun)
-
-      case JobRunStatus.Failed  => becomeFailing(jobRun)
+    val startingDeadline = computeStartingDeadline
+    if (startingDeadline.exists(d => d.toSeconds <= 0)) {
+      log.info(s"StartingDeadline reached for job ${jobRun.id}, aborting job run.")
```

Comment:
These log messages are super important IMO...   I would prefer 1) to be consistent and 2) they are as helpful as possible and 3) they provide a clear message as to "when" in the order of events this event occurred.

For this one:
how about `s"StartingDeadline timeout of ${startingDeadline} sec expired at PreStart for JobRun ${jobRun.id} created at ${jobRun.createdAt}"`
aborting should log it's own message
this way we know with 1 log message:
1) the deadline was negative or 0 and what value
2) when the job was created
3) what the id was

Reply:
thanks! hopefully better now